src/Chapter3.hs

```diff
@@ -971,6 +1172,36 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay 
+    = Monday 
+    | Tuesday 
+    | Wednesday 
+    | Thursday 
+    | Friday 
+    | Saturday
+    | Sunday 
+    deriving (Show, Eq, Enum, Bounded)
+
+
+isWeekend :: WeekDay -> Bool
+isWeekend day =
+    case day of
+        Sunday -> True
+        Saturday -> True
+        _ -> False
+
+
+nextDay :: WeekDay -> WeekDay
+nextDay day
+    | day == maxBound = minBound
+    | otherwise = succ day
+
+
+daysToParty :: WeekDay -> Int
+daysToParty day =
+    length (enumFromTo day Sunday) - 2
```

Comment:
Interesting solution!

However, it would be faster if you use `mod` function instead of calculating the length of days.